components/table/Table.razor

```diff
@@ -146,6 +146,7 @@ RenderFragment<(Table<TItem> table, IEnumerable<TItem> showItems, int level)> bo
         var table = context.table;
         var showItems = context.showItems;
         var level = context.level;
+        var count = Math.Min(showItems.Count(), table._dataSourceCount);
```

Comment:
Well, when you move reload code to ShouldRender method, here is no need to get the minimum count anymore, they are the same.

Reply:
Yep...You are right. We need a new PR for that...